src/SourcesCommand.cc

```diff
@@ -130,6 +141,197 @@ struct DirExistsCache {
   }
 };
 
+struct DebugDirs {
+  vector<string> debug_file_directories;
+  vector<string> source_directories;
+};
+
+class DebugDirManager {
```

Comment:
A comment describing what this class does would be helpful. E.g. pointing out that it controls a Python subprocess.